script.ts

```diff
@@ -0,0 +1,173 @@
+const button: HTMLButtonElement = document.querySelector(".add")!;
+const form: HTMLFormElement = document.querySelector("#adding-items")!;
+const select: HTMLSelectElement = form.querySelector("select")!;
+const quickButtons: HTMLButtonElement[] = Array.from(document.querySelectorAll(".quick-add button"));
+let localStoredItems: string | null = localStorage.getItem("items");
+let items: ItemObject[] = localStoredItems ? JSON.parse(localStoredItems) : [];
+let index: number = 0; 
+let list: HTMLElement | null = document.querySelector("#shopping-list") || null;
+let removeListButton: HTMLButtonElement | null;
+
+interface ItemObject {
+    name: string;
+    category: string;
+    id: number;
+    isChecked: boolean;
+}
+
+class ItemObject {
+    constructor (name: string, category: string) {
+        this.name = name;
+        this.category = category;
+        this.id = index;
+        this.isChecked = false;
+    }
+}
```

Comment:
I would pass the types directly into the class. 

```suggestion
class ItemObject {
    name: string;
    category: string;
    id: number;
    isChecked: boolean;
    constructor (name: string, category: string) {
        this.name = name;
        this.category = category;
        this.id = index;
        this.isChecked = false;
    }
}
```

Reply:
I'm not sure if I combine these parts of the code, will I still have type/interface "ItemObject" defined? Or maybe defining an interface is not needed when using a class?